### PR TITLE
fix(trends): Fix trends filters and statsperiod between tabs

### PR DESCRIPTION
**Pull request by k-fish**
### Summary
Since the default page change, trends were missing the default filters we had applied. This was moved into the constructor to update browserHistory when it loads (along with always setting view to avoid having to do beta feature checks).

This also fixes default periods as a result of the default page change. Since the tabs have different default periods, this adds a check to update the global selection header to make sure the period stays as default if the stats period isn't set or it's set to the same period as the page you are navigating to

### CODE REVIEW

File: src/sentry/static/sentry/app/views/performance/landing.tsx [hunk 1]
```diff
@@ -181,30 +189,30 @@ class PerformanceLanding extends React.Component<Props, State> {
...
-      }
+    const currentView = location.query.view;
 
-      newQuery.query = stringifyQueryObject(conditions);
+    const newDefaultPeriod =
+      viewKey === FilterViews.TRENDS ? DEFAULT_TRENDS_STATS_PERIOD : DEFAULT_STATS_PERIOD;
+
+    const hasStartAndEnd = newQuery.start && newQuery.end;
+
+    if (isStatsPeriodDefault(statsPeriod, newDefaultPeriod) && !hasStartAndEnd) {
```
Comment (markstory) [inline src/sentry/static/sentry/app/views/performance/landing.tsx:202]: If you put `!hasStartAndEnd` first you'll save a function call 🏌️

File: src/sentry/static/sentry/app/views/performance/trends/content.tsx [hunk 2]
```diff
@@ -28,7 +31,42 @@ type State = {
 };
 
 class TrendsContent extends React.Component<Props, State> {
+  constructor(props) {
+    super(props);
```
Comment (markstory) [inline src/sentry/static/sentry/app/views/performance/trends/content.tsx:35]: You'll want to spread args down as context comes in the constructor too.

File: src/sentry/static/sentry/app/views/performance/trends/content.tsx [hunk 3]
```diff
@@ -28,7 +31,42 @@ type State = {
...
+    props.eventView.query = query;
+
+    browserHistory.push({
+      pathname: location.pathname,
+      query: {
+        ...location.query,
+        cursor: undefined,
+        query: String(query).trim() || undefined,
+        view: FilterViews.TRENDS,
+      },
```
Comment (markstory) [inline src/sentry/static/sentry/app/views/performance/trends/content.tsx:64]: I'm guessing you want to have the default query string values in the URL bar? Redirecting from a constructor is unconventional and might not stop the first render.

Comment (k-fish) [reply src/sentry/static/sentry/app/views/performance/trends/content.tsx:64]: it won't stop the first render, but it does push correctly, that's why I'm doing `props.eventView.query = query;` to make sure the eventview also has these appended tags.

Comment (k-fish) [reply src/sentry/static/sentry/app/views/performance/trends/content.tsx]: Fixed to use intermediate component like we talked about 👍

**Review by markstory — APPROVED** [review]

**Pull request merged**